docs/notebooks/flax_basics.ipynb

```diff
@@ -0,0 +1,911 @@
+{
```

Comment:
<blockquote>ut as mentionned above, it won't be able to use shape inference as Flax is using lazy initialization at first call site.</blockquote><p>In this case with our current API you're right. If you define parameters during setup then you can't use shape inference. But it's not like with submodules.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google/flax/pull/542/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

Reply:
<p>Didn't understand the distinction with submodules ? </p>

Reply:
<p>Remove as mentionned above (diff from submodule)</p>